apps/guardian-ui/src/types.tsx

```diff
@@ -123,3 +134,37 @@ export type SetupAction =
       type: SETUP_ACTION_TYPE.SET_OUR_CURRENT_ID;
       payload: number;
     };
+
+interface SharedApiInterface {
+  auth: () => Promise<void>;
+  status: () => Promise<StatusResponse>;
+  getVerifyConfigHash: () => Promise<PeerHashMap>;
+}
+
+export interface SetupApiInterface extends SharedApiInterface {
+  setPassword: (password: string) => Promise<void>;
+  setConfigGenConnections: (
+    ourName: string,
+    leaderUrl?: string
+  ) => Promise<void>;
+  getDefaultConfigGenParams: () => Promise<ConfigGenParams>;
+  getConsensusConfigGenParams: () => Promise<ConsensusState>;
+  setConfigGenParams: (params: ConfigGenParams) => Promise<void>;
+  runDkg: () => Promise<void>;
+  verifiedConfigs: () => Promise<void>;
+  startConsensus: () => Promise<void>;
+}
+
+export interface AdminApiInterface extends SharedApiInterface {
+  version: () => Promise<Versions>;
+  fetchBlockCount: () => Promise<number>;
+  inviteCode: () => Promise<string>;
+  config: (connection: string) => Promise<ConfigResponse>;
+  audit: () => Promise<AuditSummary>;
+  modulesConfig: () => Promise<ModulesConfigResponse>;
+  moduleApiCall: <T>(moduleId: number, rpc: LightningModuleRpc) => Promise<T>;
+}
```

Comment:
One of the things I was hoping to get rid of here was the double typing of every method. How would you feel about replacing this with `Pick`ing the methods off the `GuardianApi`?
```suggestion
type SharedApiMethods = 'auth' | 'status' | 'getVerifyConfigHash';
type SetupApiMethods =
  | 'setPassword'
  | 'setConfigGenConnections'
  | 'getDefaultConfigGenParams'
  | 'getConsensusConfigGenParams'
  | 'setConfigGenParams'
  | 'runDkg'
  | 'verifiedConfigs'
  | 'startConsensus';
type AdminApiMethods =
  | 'version'
  | 'fetchBlockCount'
  | 'inviteCode'
  | 'config'
  | 'audit'
  | 'modulesConfig'
  | 'moduleApiCall';

type SharedApiInterface = Pick<GuardianApi, SharedApiMethods>;

export type SetupApiInterface = SharedApiInterface &
  Pick<GuardianApi, SetupApiMethods>;
export type AdminApiInterface = SharedApiInterface &
  Pick<GuardianApi, AdminApiMethods>;
```

You could then remove `implements SetupApiInterface, AdminApiInterface` from the `class GuardianApi` and now it's the single source of truth for method implementation.

Reply:
Oh wait even better, move the `SharedRpc`, `SetupRpc`, and `AdminRpc` enums into types and just `Pick` from those. Super single source of truth!

Reply:
Nah, you can do `keyof typeof Enum` instead of `Enum` to get the camel cased versions that we use.

Reply:
ack. did a refresh of PR and applied type defs using enums